test/unit/app/controllers/metamask-controller-test.js

```diff
@@ -35,9 +35,33 @@ const ExtensionizerMock = {
   },
 }
 
+let loggerMiddlewareMock
+const initializeMockMiddlewareLog = () => {
+  loggerMiddlewareMock = {
+    requests: [],
+    responses: [],
+  }
+}
+const tearDownMockMiddlewareLog = () => {
+  loggerMiddlewareMock = undefined
+}
+
+const createLoggerMiddlewareMock = () => (req, res, next) => {
+  if (loggerMiddlewareMock) {
+    loggerMiddlewareMock.requests.push(req)
+    next((cb) => {
+      loggerMiddlewareMock.responses.push(res)
+      cb()
+    })
+  } else {
+    next()
+  }
+}
+
```

Comment:
This mock was added to get around a difficult test failure.

We were relying on the lack of validation in `eth-json-rpc-middleware` to test a bogus transaction. This mock allows us to arbitrarily inspect RPC requests sent during tests. I'm open to moving it elsewhere, but couldn't think of a better solution.